localstack/services/opensearch/provider.py

```diff
@@ -400,6 +406,51 @@ def get_store(region: str = None) -> OpenSearchStore:
             ), "The region was not given and it could not be extracted from a request context."
         return opensearch_stores[get_aws_account_id()][region]
 
+    def accept_state_visitor(self, visitor: StateVisitor):
+        visitor.visit(opensearch_stores)
+        visitor.visit(AssetDirectory(os.path.join(config.dirs.data, "opensearch")))
+        visitor.visit(AssetDirectory(os.path.join(config.dirs.data, "elasticsearch")))
+
+    def on_after_state_load(self):
+        """Starts clusters whose metadata has been restored."""
+        for account_id, region_bundle in opensearch_stores.items():
+            for region, store in region_bundle.items():
+                for domain_name, domain_status in store.opensearch_domains.items():
+                    domain_key = DomainKey(domain_name, region, account_id)
+                    if cluster_manager().get(domain_key.arn):
+                        # cluster already restored in previous call to on_after_state_load
+                        continue
+
+                    LOG.info(f"Restoring domain {domain_name} in region {region}.")
+                    try:
+                        preferred_port = None
+                        if config.OPENSEARCH_ENDPOINT_STRATEGY == "port":
+                            # try to parse the previous port to re-use it for the re-created cluster
+                            if "Endpoint" in domain_status:
+                                preferred_port = urlparse(
+                                    f"http://{domain_status['Endpoint']}"
+                                ).port
+
+                        engine_version = domain_status.get("EngineVersion")
+                        domain_endpoint_options = domain_status.get("DomainEndpointOptions", {})
+                        create_cluster(
+                            domain_key, engine_version, domain_endpoint_options, preferred_port
+                        )
```

Comment:
`preferred_port` is used as the 4th positional parameter (instead of the 5th) here which should actually be `security_options`.

```suggestion
                        security_options = SecurityOptions.from_input(domain_status.get("AdvancedSecurityOptions"))
                        create_cluster(
                            domain_key, engine_version, domain_endpoint_options, security_options, preferred_port
                        )
```

function definition:

```python
def create_cluster(
    domain_key: DomainKey,
    engine_version: str,
    domain_endpoint_options: Optional[DomainEndpointOptions],
    security_options: Optional[SecurityOptions],
    preferred_port: Optional[int] = None,
):
```

Reply:
nice catch